pkg/apis/crd/v1beta1/types.go

```diff
@@ -151,3 +151,267 @@ type ControllerCondition struct {
 	// Human readable message indicating details
 	Message string `json:"message,omitempty"`
 }
+
+type TraceflowPhase string
+
+const (
+	// Pending is not used anymore
+	Pending   TraceflowPhase = "Pending"
+	Running   TraceflowPhase = "Running"
+	Succeeded TraceflowPhase = "Succeeded"
+	Failed    TraceflowPhase = "Failed"
+)
+
+type TraceflowComponent string
+
+const (
+	ComponentSpoofGuard    TraceflowComponent = "SpoofGuard"
+	ComponentLB            TraceflowComponent = "LB"
+	ComponentRouting       TraceflowComponent = "Routing"
+	ComponentNetworkPolicy TraceflowComponent = "NetworkPolicy"
+	ComponentForwarding    TraceflowComponent = "Forwarding"
+	ComponentEgress        TraceflowComponent = "Egress"
+)
+
+type TraceflowAction string
+
+const (
+	ActionDelivered TraceflowAction = "Delivered"
+	ActionReceived  TraceflowAction = "Received"
+	ActionForwarded TraceflowAction = "Forwarded"
+	ActionDropped   TraceflowAction = "Dropped"
+	ActionRejected  TraceflowAction = "Rejected"
+	// ActionForwardedOutOfOverlay indicates that the packet has been forwarded out of the network
+	// managed by Antrea. This indicates that the Traceflow request can be considered complete.
+	ActionForwardedOutOfOverlay TraceflowAction = "ForwardedOutOfOverlay"
+	ActionMarkedForSNAT         TraceflowAction = "MarkedForSNAT"
+	ActionForwardedToEgressNode TraceflowAction = "ForwardedToEgressNode"
+)
+
+// List the supported protocols and their codes in traceflow.
+// According to code in Antrea agent and controller, default protocol is ICMP if protocol is not inputted by users.
+const (
+	ICMPProtocolNumber int32 = 1
+	IGMPProtocolNumber int32 = 2
+	TCPProtocolNumber  int32 = 6
+	UDPProtocolNumber  int32 = 17
+	SCTPProtocolNumber int32 = 132
+)
+
+var SupportedProtocols = map[string]int32{
+	"TCP":  TCPProtocolNumber,
+	"UDP":  UDPProtocolNumber,
+	"ICMP": ICMPProtocolNumber,
+}
+
+var ProtocolsToString = map[int32]string{
+	TCPProtocolNumber:  "TCP",
+	UDPProtocolNumber:  "UDP",
+	ICMPProtocolNumber: "ICMP",
+	IGMPProtocolNumber: "IGMP",
+	SCTPProtocolNumber: "SCTP",
+}
+
+// List the supported destination types in traceflow.
+const (
+	DstTypePod     = "Pod"
+	DstTypeService = "Service"
+	DstTypeIPv4    = "IPv4"
+)
+
+var SupportedDestinationTypes = []string{
+	DstTypePod,
+	DstTypeService,
+	DstTypeIPv4,
+}
+
+// List the ethernet types.
+const (
+	EtherTypeIPv4 uint16 = 0x0800
+	EtherTypeIPv6 uint16 = 0x86DD
+)
```

Comment:
I don't know if these truly belong in this file given that they are not used by the API. I am actually not sure that these constants are used anywhere in the Antrea codebase.

Reply:
Yes, they are not used anywhere, I removed them both in v1alpha1 and v1beta1.